CodeQL query to check for insecure MaxLengthRequest values in ASP.NET applications

cldrn opened the pull request:
The directive `maxRequestLength` sets the limit for the input stream buffering threshold in KB. Attackers can use large requests to cause denial-of-service attacks.

This PR includes a CodeQL query to detect applications with larger values than the recommended one:
https://docs.microsoft.com/en-us/dotnet/api/system.web.configuration.httpruntimesection.maxrequestlength?view=netframework-4.8

hvitved commented:
Closing this PR, as it is a duplicate of https://github.com/Semmle/ql/pull/2200.

cldrn commented:
Closed the other issue as I removed that local copy and started doing each PR in their own branch. At this point I have deleted the original branch for that PR. I've addressed the comments in there but can we keep working on this one?

The other PR was on master branch which became a mess when I added others there by mistake. Hence, why I started from stratch with each PR in their own branch.

hvitved commented:
OK, let's close the other PR, and continue with this one.

hvitved commented on csharp/ql/src/Security Features/CWE-016/ASPNetMaxRequestLength.ql:
```diff
@@ -0,0 +1,18 @@
+/**
+ * @name Large maxRequestLength value
```
Add single quotes around `maxRequestLength`.

cldrn replied:
Fixed.

hvitved commented on csharp/ql/src/Security Features/CWE-016/ASPNetMaxRequestLength.ql:
```diff
@@ -0,0 +1,18 @@
+ * @description Setting a large 'maxRequestLength' value may render a webpage vulnerable to
+ *              denial-of-service attacks.
+ * @kind problem
+ * @problem.severity warning
```
Add
```
 * @id cs/web/large-max-request-length
 * @tags security
 *       frameworks/asp.net
 *       external/cwe/cwe-16
```
(from the other PR)

cldrn replied:
Fixed.

cldrn commented:
Addressed the pending issues. 

Thanks for the feedback. Are security test cases needed for all codeql queries?

hvitved commented:
> Are security test cases needed for all codeql queries?

Yes, we like to have test cases for all queries.

hvitved commented:
Thanks again for your contribution 🎉

cldrn commented:
Thanks for the help getting ready. These were from a real sca I had to do so I thought it would be a great way to get familiar with the framework.
